### PR TITLE
feat: Allow setting remote network by name

### PR DESCRIPTION
**Pull request by ekampf**
## Changes

When configuring the operator, allow setting the remote network as a reference by name.

### CODE REVIEW

File: deploy/golden_test.go [hunk 1]
```diff
@@ -46,21 +46,25 @@ func TestHelmRender(t *testing.T) {
     for _, f := range files {
         if !f.IsDir() && strings.HasSuffix(f.Name(), ".yaml") && !strings.HasSuffix(f.Name(), ".golden.yaml") {
             // Render this values.yaml file
-            output := helm.RenderTemplate(t,
+            output, error := helm.RenderTemplateE(t,
```
Comment (ekampf) [inline deploy/golden_test.go:49]: I enhanced `golden_test.go` to also print error information so we can test for error cases too
(useful for testing schema validation etc)

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -20,10 +21,15 @@ repos:
         exclude: >
           (?x)^(
             deploy/twingate-operator/templates/.*|
+            deploy/test/golden/.*|
```
Comment (ekampf) [inline .pre-commit-config.yaml:24]: In case of failure the ".golden.yaml" file isnt technically a yaml

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -20,10 +21,15 @@ repos:
...
           )$
       - id: debug-statements
       - id: destroyed-symlinks
       - id: end-of-file-fixer
+        exclude: >
+          (?x)^(
+            deploy/test/golden/.*|
```
Comment (ekampf) [inline .pre-commit-config.yaml:31]: We dont want fixes or golden test files - if pre-commit changes them, the test will fail

**Review by liorr — APPROVED** [review]
LGTM

**Pull request merged**